### PR TITLE
runtime-rs: return the error of kill_process

### PR DESCRIPTION
**Pull request by liubin**
`.ok()` will convert Result to Option, if there is an error,
the error will not be returned to the callers.

Fixes: #5018

Signed-off-by: Bin Liu <bin@hyper.sh>

### CODE REVIEW

**Review by studychao — APPROVED** [review]
LGTM

**Review by jodh-intel — APPROVED** [review]
Nice - thanks @liubin.

lgtm

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs [hunk 1]
```diff
@@ -136,16 +136,25 @@ impl ContainerManager for VirtContainerManager {
         let c = containers
             .get(container_id)
             .ok_or_else(|| Error::ContainerNotFound(container_id.clone()))?;
+
+        // skip stopped containers
+        if c.is_stopped().await {
+            return Ok(());
+        }
```
Comment (gkurz) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs:143]: Why is this being added ? Is it fixing an actual bug ? Or is it because we expect kill_process() to return an error if the container is stopped ? Something else ? In any case, please add some more details in the changelog or comments.

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs [hunk 2]
```diff
@@ -253,7 +262,7 @@ impl ContainerManager for VirtContainerManager {
         let c = containers
             .get(container_id)
             .ok_or_else(|| Error::ContainerNotFound(container_id.to_string()))?;
-        c.update(&resource).await.context("stats")
+        c.update(&resource).await.context("update_container")
```
Comment (gkurz) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs:265]: This fix is minor but it doesn't seem to belong to this PR.

File: src/runtime-rs/crates/service/src/manager.rs [hunk 3]
```diff
@@ -65,7 +65,7 @@ async fn send_event(
             &namespace,
         ])
         .spawn()
-        .context("sawn cmd")?;
+        .context("spawn containerd cmd")?;
```
Comment (gkurz) [inline src/runtime-rs/crates/service/src/manager.rs:68]: Ditto

**Comment by studychao** [comment]
@liubin Hi, any update on this?

**Comment by liubin** [comment]
@studychao I'll split some typo/name fixes into another new one, and about the kill_process, I'll rethink it asap.
